### ISSUE TITLE
ENH: Add neighboring DWI correlation QC metric

### ISSUE THREAD

**Issue by mattcieslak** [dipy/dipy#3154]
## Description

We've found the Neighboring DWI Correlation to be very useful as an automated QC metric. It's currently only implemented in DSI Studio, but is easy to calculate and would be a great addition to DIPY. 

Some prior discussion [here](https://github.com/nipreps/mriqc/pull/1131).

**Comment by skoudoro** [comment]
Hi @mattcieslak, 

Looking forward for this addition !

**Comment by araikes** [comment]
@mattcieslak... Did you see the new DW contrast metric Frank just added? Ratio of NDC and orthogonal DWI correlation

**Comment by arokem** [comment]
@araikes : that sounds interesting! Is there a paper that describes this metric? Is it also primarily used for QC? Any clear benefits over NDC?  Would it be easy to add to this module (in a subsequent PR)? How is "orthogonal DWI correlation" calculated?

**Comment by mattcieslak** [comment]
> @mattcieslak... Did you see the new DW contrast metric Frank just added? Ratio of NDC and orthogonal DWI correlation

yes, I think these are really clever and am going to try them out as I recompute NDC on all my other data

**Comment by araikes** [comment]
@arokem, honestly I know very little of the details. In theory it's a QC metric. I saw an X post from Frank about it and that's the extent (https://twitter.com/FangChengYeh/status/1773160062627783158). Still need to dig in but I knew that @mattcieslak was working on implementing the NDC metric here
